logos/logos-agent/app/auth.py

```
@@ -0,0 +1,137 @@
+            token,
+            signing_key.key,
+            algorithms=["RS256", "RS384", "RS512", "ES256", "ES384"],
+            audience=settings.keycloak_audience or None,
+            issuer=settings.keycloak_issuer_uri or None,
+            options={
+                "verify_aud": bool(settings.keycloak_audience),
+                "verify_iss": bool(settings.keycloak_issuer_uri),
+            },
+        )
+    except jwt.ExpiredSignatureError:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Token expired")
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)